src/jit/codegenarm64.cpp

```diff
@@ -3409,11 +3411,38 @@ void CodeGen::genCodeForStoreInd(GenTreeStoreInd* tree)
 
         if (tree->gtFlags & GTF_IND_VOLATILE)
         {
-            // issue a full memory barrier a before volatile StInd
-            instGen_MemoryBarrier();
+            bool useStoreRelease = (dataReg >= REG_INT_FIRST) && (dataReg <= REG_ZR) && !addr->isContained() &&
+                                   ((attr == EA_1BYTE) || !(tree->gtFlags & GTF_IND_UNALIGNED));
```

Comment:
Looks like ```genIsValidIntReg(regNumber reg)``` is a direct replacement.  I'll use that unless you strongly prefer not float.

> It also should not be possible to have the case of an "unaligned" 1 byte load/store.
A check could be added when we decide to set that flag.

My original code added an assert unaligned not set on byte size code which triggered on some of the directed volatile prefix tests.

Since Unaligned is a IL prefix, I think we do not remove the prefix in import.  I expect the directed volatile prefix tests set the unaligned flag on byte sized load/store.  We can add code to clear the flag during import for byte sized ops.  I do not think the prefix on a byte load/store is explicitly illegal IL, so ignoring it would be the best course of action IMO.

Either way I'll remove the special case here.